storybook/src/components/shared/decorators.tsx

```diff
@@ -5,6 +5,13 @@
 
 import type { StoryContext, StoryFn } from '@storybook/react'
 
+/** Applies a light background colour if the `contrastColor` arg is true. */
+export const contrastColorDecorator = (Story: StoryFn, context: StoryContext) => (
+  <div className={context.args['contrastColor'] ? 'ams-docs-light-background' : undefined}>
+    <Story />
+  </div>
+)
+
 /** Applies a dark background colour if the `inverseColor` arg is true. */
 export const inverseColorDecorator = (Story: StoryFn, context: StoryContext) => (
   <div className={context.args['inverseColor'] ? 'ams-docs-dark-background' : undefined}>
```

Comment:
Can we increase the specificity here? Having both `contrastColor` and `inverseColor` selected, the components currently show the inverse color on a light background.

Reply:
The specificity was already in place through source order. The order in which Storybook applies the background decorators didn’t match – now they do.

I’ve also documented not to use both props together and verified that the ‘inverse’ appearance wins if one dows.